src/main/java/org/jfrog/bamboo/configuration/util/TaskConfigurationValidations.java

```diff
@@ -0,0 +1,122 @@
+package org.jfrog.bamboo.configuration.util;
+
+import com.atlassian.bamboo.collections.ActionParametersMap;
+import com.atlassian.bamboo.utils.error.ErrorCollection;
+import org.apache.commons.configuration.ConversionException;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jfrog.bamboo.admin.ServerConfig;
+import org.jfrog.bamboo.admin.ServerConfigManager;
+import org.jfrog.bamboo.context.AbstractBuildContext;
+import org.jfrog.bamboo.release.vcs.VcsTypes;
+
+/**
+ * Created by Bar Belity on 10/03/2020.
+ */
+public class TaskConfigurationValidations {
+
+    /**
+     * Validate that an executable is provided.
+     */
+    public static void validateExecutable(String executableKey, @NotNull ActionParametersMap params, @NotNull ErrorCollection errorCollection) {
+        if (StringUtils.isBlank(params.getString(executableKey))) {
+            errorCollection.addError(executableKey, "Please specify an Executable.");
+        }
+    }
+
+    /**
+     * Validate that a build-jdk is provided.
+     */
+    public static void validateJdk(String buildJdkKey, @NotNull ActionParametersMap params, @NotNull ErrorCollection errorCollection) {
+        if (StringUtils.isBlank(params.getString(buildJdkKey))) {
+            errorCollection.addError(buildJdkKey, "Please specify Build JDK.");
+        }
+    }
+
+    /**
+     * Validate that the provided server ID is valid.
+     */
+    public static void validateArtifactoryServer(String serverKey, ServerConfigManager serverConfigManager, @NotNull ActionParametersMap params, @NotNull ErrorCollection errorCollection) {
+        if (!params.containsKey(serverKey)) {
+            return;
+        }
+        long configuredServerId = getConfiguredServerId(serverKey, params);
+        if (configuredServerId == -1) {
+            return;
+        }
+        validateServerConfigExists(configuredServerId, serverKey, serverConfigManager, errorCollection);
+    }
+
+    /**
+     * Validate that the provided server ID and the corresponding repository are valid.
+     */
+    public static void validateArtifactoryServerAndRepo(String serverKey, String repositoryKey, ServerConfigManager serverConfigManager, @NotNull ActionParametersMap params, @NotNull ErrorCollection errorCollection) {
+        if (!params.containsKey(serverKey)) {
+            return;
+        }
+        long configuredServerId = getConfiguredServerId(serverKey, params);
+        if (configuredServerId == -1) {
+            return;
+        }
+        validateServerConfigExists(configuredServerId, serverKey, serverConfigManager, errorCollection);
```

Comment:
It looks like the above code lines (since the beginning of this method) are duplicated. We can probably call the *validateArtifactoryServer* method instaed.

Reply:
Created a new private method in TaskConfigurationValidations -> validateServer.
The public methods will use it.